lib/classes/Swift/Signers/DKIMSigner.php

```diff
@@ -328,14 +361,21 @@ public function setHashAlgorithm($hash)
      *
      * @param string $canon
      *
+     * @throws Swift_SwiftException
+     *
      * @return $this
      */
     public function setBodyCanon($canon)
     {
-        if ($canon == 'relaxed') {
-            $this->_bodyCanon = 'relaxed';
-        } else {
-            $this->_bodyCanon = 'simple';
+        switch ($canon) {
+            case 'simple':
+                $this->_bodyCanon = 'simple';
+                break;
+            case 'relaxed':
+                $this->_bodyCanon = 'relaxed';
+                break;
+            default:
+                throw new Swift_SwiftException('Unable to set the body canon, must be one of simple or relaxed (%s given).', $canon);
```

Comment:
I think this has lower risk for incompatibility. `setSignatureTimestamp()` would be more difficult, because in actual code it accepts strings. In my pr it do only accept bool or int.